docs/reference/query-languages/esql/esql-commands.md

```diff
@@ -663,6 +663,51 @@ FROM employees
 | LIMIT 5
 ```
 
+## `LOOKUP JOIN` [esql-lookup-join]
+
+`LOOKUP JOIN` is useful for any scenario where you need to pull in information from a lookup index to streamline data enrichment and analysis.
+
+**Syntax**
+
+```esql
+FROM firewall_logs
+| LOOKUP JOIN threat_list ON source.IP
+```
+
+**Parameters**
+
+TBD
+
+**Description**
+
+TBD
+
+**Examples**
+
+**IP Threat correlation**: This query would allow you to see if any source IPs match known malicious addresses.
+
+```esql
+FROM firewall_logs
+| LOOKUP JOIN threat_list ON source.IP
```

Comment:
super-nit: to truly match the example description, maybe it'd be useful to add something like
```
| WHERE threat_level IS NOT NULL
```
We do perform a lookup in the threat list, but we don't do anything with the result. Checking for a field from the `threat_list` to be not null implies that the `source.IP` was actually in the threat_list in the first place. LOOKUP JOIN doesn't automatically discard rows that have no match in the lookup index. (The latter behavior would be an INNER JOIN, while LOOKUP JOIN behaves like a LEFT OUTER JOIN.)

Reply:
do we want to specify this clear guidance of how lookup join works .. as a left outer join. I think this would probably be beneficial to folks if it does mimic that. We can use common language to explain.. for instance

`LOOKUP JOIN` works similar to a LEFT OUTER join in SQL. It will provide all the rows from the input table and if there is matching data in the lookup index that will be provided as well. If there is no match, the rows from the input table will still be provided and any missing data will be returned as `null`. I am not sure if this is accurately explaining it so please keep me honest. But that is how left outer joins work.